From e8c3866db0c6d0d263e4ca8217d3baed16f364cf Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Tue, 24 Jun 2025 05:48:27 -0700
Subject: [PATCH] winex11: HACK: Reduce the amount of
 update_gl_drawable_{size,offscreen} calls.

Temporary performance optimization hack until Remi finishes refactoring.
---
 dlls/winex11.drv/init.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/dlls/winex11.drv/init.c b/dlls/winex11.drv/init.c
index e6b6213f289..67fa2286793 100644
--- a/dlls/winex11.drv/init.c
+++ b/dlls/winex11.drv/init.c
@@ -381,13 +381,17 @@ static void X11DRV_client_surface_present( struct client_surface *client, HDC hd
 {
     struct x11drv_client_surface *surface = impl_from_client_surface( client );
     HWND hwnd = client->hwnd, toplevel = NtUserGetAncestor( hwnd, GA_ROOT );
+    static unsigned char drawable_update_throttle = 0;
     struct x11drv_win_data *data;
     RECT rect_dst, rect;
     Drawable window;
     HRGN region;
 
-    client_surface_update_size( hwnd, surface );
-    client_surface_update_offscreen( hwnd, surface );
+    if (!drawable_update_throttle++)
+    {
+        client_surface_update_size( hwnd, surface );
+        client_surface_update_offscreen( hwnd, surface );
+    }
 
     if (!hdc) return;
     window = X11DRV_get_whole_window( toplevel );
-- 
2.50.0

